Sources/PrimerSDK/Classes/Services/API/Primer/PrimerAPIClient+3DS.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  PrimerAPIClient+3DS.swift
+//  PrimerSDK
+//
+//  Created by Evangelos Pittas on 2/4/21.
+//
+
+import Foundation
+
+extension PrimerAPIClient {
+    
+    func threeDSecureBeginAuthentication(clientToken: DecodedClientToken, paymentMethodToken: PaymentMethodToken, threeDSecureBeginAuthRequest: ThreeDS.BeginAuthRequest, completion: @escaping (_ result: Result<ThreeDS.BeginAuthResponse, Error>) -> Void) {
+        let endpoint = PrimerAPI.threeDSecureBeginAuthentication(clientToken: clientToken, paymentMethodToken: paymentMethodToken, threeDSecureBeginAuthRequest: threeDSecureBeginAuthRequest)
+        networkService.request(endpoint) { (result: Result<ThreeDS.BeginAuthResponse, NetworkServiceError>) in
+            switch result {
+            case .success(let threeDSecureBeginAuthResponse):
+                completion(.success(threeDSecureBeginAuthResponse))
+            case .failure(let error):
+                ErrorHandler.shared.handle(error: error)
+                completion(.failure(PrimerError.threeDSFailed))
+            }
+        }
+    }
+    
+    func threeDSecureStatus(clientToken: DecodedClientToken, url: String, completion: @escaping (_ result: Result<ThreeDS.BeginAuthResponse, Error>) -> Void) {
+        let endpoint = PrimerAPI.threeDSecureStatus(clientToken: clientToken, url: url)
+        networkService.request(endpoint) { (result: Result<ThreeDS.BeginAuthResponse, NetworkServiceError>) in
+            switch result {
+            case .success(let authResponse):
+                print(authResponse)
+                Timer.scheduledTimer(withTimeInterval: 3, repeats: false) { (_) in
+                    self.threeDSecureStatus(clientToken: clientToken, url: url, completion: completion)
+                }
```

Comment:
Well, not quite. Long polling is not needed on the native side. I added a small delay before making the post auth request, just to be on the safe side. Because if Netcetera hadn't had the time to mark with pass or fail, our post auth request will fail.